src/views/Course.vue

```diff
@@ -0,0 +1,168 @@
+                    <v-col cols="1" class="assessment-item">
+                      {{ item.grade ? item.grade : 'N/A' }}
+                    </v-col>
+                    <v-col cols="7" class="assessment-item">
+                      {{ item.type }}: {{ item.description }}
+                    </v-col>
+                    <v-col cols="1" class="text--secondary assessment-item">
+                      <v-fade-transition leave-absolute>
+                        <span>
+                          {{ item.weight }}
+                        </span>
+                      </v-fade-transition>
+                    </v-col>
+                    <v-col cols="2" class="text--secondary assessment-item">
+                      <v-fade-transition leave-absolute>
+                        <span>
+                          {{
+                            item.deadline
+                              ? item.deadline
+                              : item.on_going
+                              ? 'On-going'
+                              : 'TBD'
+                          }}
+                        </span>
+                      </v-fade-transition>
+                    </v-col>
+                    <v-col cols="1">
+                      <v-fade-transition leave-absolute>
+                        <span>
+                          <v-btn icon>
+                            <v-icon class="mr-1">
+                              mdi-square-edit-outline
+                            </v-icon>
+                          </v-btn>
+                          <v-btn icon>
+                            <v-icon class="mr-1"> mdi-delete </v-icon>
+                          </v-btn>
+                        </span>
+                      </v-fade-transition>
+                    </v-col>
+                  </v-row>
+                </template>
+              </v-expansion-panel-header>
+              <v-expansion-panel-content>
+                <!-- TODO: Subtasks -->
+              </v-expansion-panel-content>
+            </v-expansion-panel>
+          </v-expansion-panels>
+        </div>
+      </v-sheet>
+    </v-col>
+  </v-row>
+</template>
+
+<script>
+import axios from 'axios';
+
+export default {
+  created() {
+    window.addEventListener('resize', this.handleResize);
+  },
+  data() {
+    return {
+      height: window.innerHeight,
+      file: null,
+      assessments: []
+    };
+  },
+  methods: {
+    handleResize() {
+      this.height = window.innerHeight;
+    },
+    onPickFile() {
+      this.$refs.fileInput.click();
+    },
+    onFilePicked(event) {
+      const { files } = event.target;
+      const filename = files[0].name;
+      const fileReader = new FileReader();
+      fileReader.addEventListener('load', () => {
+        this.fileUrl = fileReader.result;
+      });
+      fileReader.readAsDataURL(files[0]);
+      this.file = files[0];
+      if (window.confirm(`Do you want to use this syllabus: ${filename}?`)) {
```

Comment:
Maybe in the future, we can display this PDF in a visual format 🤔

Reply:
hmm wdym

Reply:
SO instead of viewing a popup, it would display the pdf to upload so that the user can check it again just in case!

Reply:
added a todo! may look into that later if we have time